File: module-services/src/main/java/eu/geekhome/services/configurable/RequiredDoubleValidator.java
package eu.geekhome.services.configurable;


import eu.geekhome.services.localization.Resource;

public class RequiredDoubleValidator implements Validator<Double> {
    @Override
    public Resource getReason() {
        return new Resource("This field is required",
                "To pole jest wymagane");
    }

    @Override
    public boolean validate(Double fieldValue) {
        return fieldValue != null;
    }
}

File: module-services/src/main/java/eu/geekhome/services/configurable/RequiredDoubleValidator.kt
package eu.geekhome.services.configurable

import eu.geekhome.services.localization.Resource

class RequiredDoubleValidator : Validator<Double?> {
    override val reason: Resource
        get() = Resource(
            "This field is required",
            "To pole jest wymagane"
        )

    override fun validate(fieldValue: Double?): Boolean {
        return fieldValue != null
    }
}